packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -68,62 +101,76 @@ class _BannerMessageContainerState extends State<BannerMessageContainer>
     final newController = Controllers.of(context)?.bannerMessages;
     if (newController == controller) return;
     controller = newController;
+  }
+
+  void push(BannerMessage message) {
+    if (controller.isMessageDismissed(message) ||
+        controller.isMessageVisible(message)) return;
+    // We push the banner message in a post frame callback because otherwise,we'd be
+    // trying to call setState while the parent widget `BannerMessages` is in the middle
+    // of `build`.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (!mounted) return;
+      setState(() {
```

Comment:
seems surprising to have a setState around a call to a controller. setState calls should be when the direct state of this widget changes. can you instead have this widget listen for when the messages associated with its screen id in the controller change?

Reply:
switched to using value listenable